internal/utils/http.go

```diff
@@ -0,0 +1,16 @@
+package utils
+
+import (
+	"net/http"
+
+	"github.com/eagraf/habitat-new/internal/node/logging"
+)
+
+var (
+	log = logging.NewLogger()
+)
+
+func LogAndHTTPError(w http.ResponseWriter, error string, code int) {
```

Comment:
imo this should take the entire `err` as well as a `debugMsg string` so we know where the error is coming from. AI should do a good job auto completing the existing callsites

Reply:
Ok i did this